Examples/Algorithms/Utilities/src/PrototracksToTracks.cpp

```diff
@@ -0,0 +1,93 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Utilities/ProtoTracksToTracks.hpp"
+
+#include "Acts/Surfaces/PerigeeSurface.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/MeasurementCalibration.hpp"
+#include "ActsExamples/EventData/ProtoTrack.hpp"
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsExamples/Utilities/EventDataTransforms.hpp"
+
+#include <algorithm>
+
+namespace ActsExamples {
+
+PrototracksToTracks::PrototracksToTracks(Config cfg, Acts::Logging::Level lvl)
+    : IAlgorithm("PrototracksToTracks", lvl), m_cfg(std::move(cfg)) {
+  m_outputTracks.initialize(m_cfg.outputTracks);
+  m_inputProtoTracks.initialize(m_cfg.inputProtoTracks);
+  m_inputMeasurements.initialize(m_cfg.inputMeasurements);
+}
+
+ProcessCode PrototracksToTracks::execute(const AlgorithmContext& ctx) const {
+  auto trackContainer = std::make_shared<Acts::VectorTrackContainer>();
+  auto mtj = std::make_shared<Acts::VectorMultiTrajectory>();
+  TrackContainer tracks(trackContainer, mtj);
+
+  PassThroughCalibrator calibratorImpl;
+  MeasurementCalibratorAdapter calibrator(calibratorImpl,
+                                          m_inputMeasurements(ctx));
+
+  auto refSurface = Acts::Surface::makeShared<Acts::PerigeeSurface>(
+      Acts::Vector3{0., 0., 0.});
+  const auto& prototracks = m_inputProtoTracks(ctx);
+  ACTS_DEBUG("Received " << prototracks.size() << " prototracks");
+
+  float avgSize = 0;
+  std::size_t minSize = std::numeric_limits<std::size_t>::max();
+  std::size_t maxSize = 0;
+
+  for (const auto& protoTrack : prototracks) {
+    if (protoTrack.empty()) {
+      continue;
+    }
+
+    avgSize += protoTrack.size();
+    minSize = std::min(minSize, protoTrack.size());
+    maxSize = std::max(maxSize, protoTrack.size());
+
+    std::size_t tip = Acts::kTrackIndexInvalid;
+    for (auto idx : protoTrack) {
+      auto trackStateProxy =
+          mtj->makeTrackState(Acts::TrackStatePropMask::Calibrated, tip);
+      tip = trackStateProxy.index();
+
+      IndexSourceLink sl(Acts::GeometryIdentifier{}, idx);
+
+      calibrator.calibrate({}, {}, Acts::SourceLink{sl}, trackStateProxy);
```

Comment:
Does this do anything? IIRC there's no predicted state to speak of, right? So the calibration will effectively be identity only. 

OTOH, we should avoid using default constructed contexts, so you might want to at least use the contexts from the event context.

Reply:
Good point with the contexts... Yeah, but without the calibration the matching to particles won't work later right?

Reply:
Not sure I follow how the matching and the calibration are connected.

Reply:
This line which is executed during calibration:
`trackState.setUncalibratedSourceLink(sourceLink);`

Of course we could skip calibration and just execute this line...